testsuite/tests/typing-gadts/pr5985.ml

```diff
@@ -78,7 +78,7 @@ let eq : (('a, 'b) Ephemeron.K1.t, ('c, 'd) Ephemeron.K1.t) eq = eq;;
 type _ t = T : 'a -> ('a, 'b) Ephemeron.K1.t t;; (* fail *)
 [%%expect{|
 type (_, _) eq = Eq : ('a, 'a) eq
-val eq : 'a = <poly>
+val eq : 'b = <poly>
```

Comment:
My understanding is that this name changed because a type variable named `'a` was created when instantiating the `Eq` subterm, and so the other type variable created to represent the return of `Obj.magic` could not be named `'a` anymore and had to be named `'b`. Or maybe it is named `'b` because we have `val magic : 'a -> 'b` in Obj ?

Reply:
I believe it's from the annotation on `magic` in `Obj`.